Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3560,6 +3560,35 @@ const NativeWebKeyboardEvent& WebPageProxy::firstQueuedKeyEvent() const
     return internals().keyEventQueue.first();
 }
 
+void WebPageProxy::sendKeyEvent(const NativeWebKeyboardEvent& event)
+{
+    m_process->recordUserGestureAuthorizationToken(event.authorizationToken());
+
+    auto handleKeyEventReply = [this, protectedThis = Ref { *this }] (std::optional<WebEventType> eventType, bool handled) {
+        if (!m_pageClient)
+            return;
+        if (!eventType)
+            return;
+        didReceiveEvent(*eventType, handled);
+    };
+
+    auto targetFrame = m_mainFrame;
+    if (m_focusedFrame) {
+        targetFrame = m_focusedFrame;
+        auto remotePageProxy = internals().domainToRemotePageProxyMap.get(RegistrableDomain(targetFrame->url()));
+        if (remotePageProxy) {
```

Comment:
if (auto remotePageProxy = m_focusedFrame.remotePageProxy())
You don't need targetFrame at all.